builder/simplify.h

```diff
@@ -29,6 +29,11 @@ bool prove_false(const expr& condition, const bounds_map& bounds = bounds_map())
 // Find the interval for `var` that makes `e` true.
 interval_expr where_true(const expr& condition, var x);
 
+// Find a maximum value of x which makes lhs <= rhs expression true. The variable x can
+// only appear in the lhs expression. The search goes backwards from initial_guess up
+// to some fixed depth.
+expr where_less_equal(const expr& lhs, const expr& rhs, var x, const expr& initial_guess);
```

Comment:
I don't think splitting the condition into lhs, rhs and requiring it be compared with <= helps much. I think maybe what you intended is to express that this function only looks for an upper bound of where x is true, given an initial guess that is greater than the upper bound, but that is separate from what kind of condition it is, so I think this function could be `where_true_upper_bound` and just take a condition (with <= baked in in this case).

Also, I think we should just delete `where_true` now that it is unused and is pretty crappy. I was also thinking of just a little local helper function in slide_and_fold_storage.cc rather than a publicly visible helper.